packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -43,31 +47,67 @@ const isActiveSidebarItem = (item, activePath) => {
   return false;
 };
 
+type CollapseMap = Record<string, boolean | undefined>;
+
+const createInitialStates = (
+  items: readonly PropSidebarItem[],
+): CollapseMap => {
+  const map: Record<string, boolean | undefined> = {};
+  items.forEach((childItem: PropSidebarItem) => {
+    const itemType = childItem.type;
+    const itemLabel = childItem.label;
+    if (itemType === 'category') {
+      map[itemLabel] = (childItem as PropSidebarItemCategory).collapsed;
+    } else {
+      map[itemLabel] = false;
+    }
+  });
+  return map;
+};
+
+const createCollapseHandler = (
+  currStates: CollapseMap,
+  setStates: (items: CollapseMap) => void,
+  currLabel: string,
+) => {
+  return (isCollapsed: boolean) => {
+    const states = {...currStates};
+    Object.keys(states).forEach((label: string) => {
+      if (currLabel === label) {
+        states[label] = isCollapsed;
+      } else if (!isCollapsed) {
+        states[label] = true;
+      }
+    });
+    setStates(states);
+  };
+};
+
 function DocSidebarItemCategory({
   item,
   onItemClick,
   collapsible,
+  collapsed,
+  setCollapsed,
   activePath,
+  autoCollapse,
   ...props
 }) {
   const {items, label} = item;
 
   const isActive = isActiveSidebarItem(item, activePath);
   const wasActive = usePrevious(isActive);
-
-  // active categories are always initialized as expanded
-  // the default (item.collapsed) is only used for non-active categories
-  const [collapsed, setCollapsed] = useState(() => {
-    if (!collapsible) {
-      return false;
-    }
-    return isActive ? false : item.collapsed;
-  });
+  const isCollapsed = isActive ? false : collapsed;
```

Comment:
I think this behavior is weird, according to seeing it in practice on the deploy preview.
If we collapse other categories on click, shouldn't we collapse all of them, not just the non-active ones?